05_debug/debug_module.c

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#include <linux/ctype.h>
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/printk.h>
+
+#include <linux/sysfs.h>
+#include <linux/proc_fs.h>
+#include <linux/device.h>
+#include <linux/uaccess.h>
+
+#include <linux/types.h>
+#include <linux/list.h>
+#include <linux/slab.h>
+
+MODULE_AUTHOR("Andrii Synenko");
+MODULE_DESCRIPTION("List data module for Linux Kernel ProCamp");
+MODULE_LICENSE("Dual BSD/GPL");
+
+#define PROCFS_BUFF_SIZE 255
+#define PROCFS_DIR_NAME	"debug_module"
+#define PROCFS_ENT_NAME	"upper"
+#define SYSFS_DIR_NAME	PROCFS_DIR_NAME
+#define SYSFS_ENT_NAME	"lower"
+#define STAT_ENT_NAME	"stat"
+
+struct convertor_struct {
+	char *str_buff;
+
+	int converted_last;
+	int converted_total;
+	int processed_last;
+	int processed_total;
+
+} sysfs_conv, procfs_conv;
```

Comment:
Use separate type declaration and two `static struct ...;`

Reply:
Fixed

Reply:
Remove ` sysfs_conv, procfs_conv` above and add `static` below.

Reply:
Fixed